src/common/user-store/user-store.ts

```diff
@@ -136,12 +136,12 @@ export class UserStore extends BaseStore<UserStoreModel> /* implements UserStore
   /**
    * Toggles the hidden configuration of a table's column
    */
-  toggleTableColumnVisibility(tableId: string, columnId: string) {
-    if (!this.hiddenTableColumns.get(tableId)) {
-      this.hiddenTableColumns.set(tableId, new ObservableToggleSet());
+  toggleTableColumnVisibility(tableId: string, columnId: string, defaultHidden: boolean) {
+    if (this.hiddenTableColumns.get(tableId)?.get(columnId) === undefined){
```

Comment:
nit: this should use `has`

Reply:
further nit, should we either warn or initialize `this.hiddenTableColumns.get(tableId)` if it is not present?

Reply:
Yes, `has` is cleaner I agree, dont know why I didnt think about it. Will update.

Second point, in theory it should be present from being set in the onMount hook, but to be safe we could log a warning, yeah.

Reply:
I think if we move to initialize, we no longer need the onmount hook at all.

Reply:
Sure, let me update that.